packages/elements-core/src/utils/ref-resolving/ReferenceResolver.ts

```diff
@@ -0,0 +1,29 @@
+import { resolveInlineRef } from '@stoplight/json';
+import { Dictionary } from '@stoplight/types';
+
+export type ReferenceInfo = {
+  source: string | null;
+  pointer: string | null;
+};
+
+export type ReferenceResolver = (ref: ReferenceInfo, propertyPath: string[] | null, currentObject: object) => any;
+
+export const defaultResolver =
+  (contextObject: object): ReferenceResolver =>
+  ({ pointer }, _, currentObject) => {
+    const activeObject = contextObject ?? currentObject;
+
+    if (pointer === null) {
+      return null;
+    }
+
+    if (pointer === '#') {
+      return activeObject;
+    }
+
+    const resolved = resolveInlineRef(activeObject as Dictionary<string>, pointer);
```

Comment:
`activeObject` is certainly not a `Dictionary<string>` (aka `Record<string, string>`). Other than a proxy, pretty much nothing *is*, right? `resolveInlineRef` wants a `Dictionary<unknown>` (aka `Record<string, unknown>` or even `Record<string, unknown | undefined>`) which raises the JS mystery, *are not-defined properties practically the same as defined ones with the value `undefined`?*

TBH it's `resolveInlineRef`'s typings that are broken. Strictly speaking it's currently saying *give me an object with all the properties defined*. We can either try to argue about that and fix it, or at least make that cast ` as Record<string, undefined>` (or you can use `Dictionary` but I think we should be preferring the built-in for newer code, for clarity)